src/game/server/neo/bot/behavior/neo_bot_command_follow.cpp

```diff
@@ -0,0 +1,374 @@
+	FCVAR_CHEAT, "Distance to compare friendly repulsion forces", true, 1, true, 100000);
+ConVar sv_neo_bot_cmdr_look_weights_wall_repulsion_whisker_dist("sv_neo_bot_cmdr_look_weights_wall_repulsion_whisker_dist", "500",
+	FCVAR_CHEAT, "Distance to extend whiskers", true, 1, true, 100000);
+
+//---------------------------------------------------------------------------------------------
+CNEOBotCommandFollow::CNEOBotCommandFollow()
+{
+}
+
+//---------------------------------------------------------------------------------------------
+ActionResult< CNEOBot >	CNEOBotCommandFollow::OnStart(CNEOBot *me, Action< CNEOBot > *priorAction)
+{
+	m_path.SetMinLookAheadDistance(me->GetDesiredPathLookAheadRange());
+
+	if (!FollowCommandChain(me))
+	{
+		return Done("No commander to follow");
+	}
+
+	return Continue();
+}
+
+//---------------------------------------------------------------------------------------------
+ActionResult< CNEOBot >	CNEOBotCommandFollow::Update(CNEOBot *me, float interval)
+{
+	if (!FollowCommandChain(me))
+	{
+		return Done("Lost commander or released");
+	}
+
+	m_path.Update(me);
+
+	return Continue();
+}
+
+//---------------------------------------------------------------------------------------------
+ActionResult< CNEOBot > CNEOBotCommandFollow::OnResume(CNEOBot *me, Action< CNEOBot > *interruptingAction)
+{
+	if (!FollowCommandChain(me))
+	{
+		return Done("No commander to follow on resume");
+	}
+
+	return Continue();
+}
+
+//---------------------------------------------------------------------------------------------
+void CNEOBotCommandFollow::OnEnd(CNEOBot *me, Action< CNEOBot > *nextAction)
+{
+	me->m_hLeadingPlayer = nullptr;
+	me->m_hCommandingPlayer = nullptr;
+}
+
+
+// ---------------------------------------------------------------------------------------------
+// Process commander ping waypoint commands for bots
+// true - order has been processed, continue following
+// false - no order, stop following
+bool CNEOBotCommandFollow::FollowCommandChain(CNEOBot* me)
+{
+	Assert(me);
+	CNEO_Player* pCommander = me->m_hCommandingPlayer.Get();
+
+	if (!pCommander)
+	{
+		return false;
+	}
+
+	// Mirror behavior of leader if we have one
+	if (me->m_hLeadingPlayer.Get())
+	{
+		CNEO_Player *pPlayerToMirror = me->m_hLeadingPlayer.Get();
+		if (pPlayerToMirror)
+		{
+			if (pPlayerToMirror->GetInThermOpticCamo())
+			{
+				me->EnableCloak(4.0f);
+			}
+			else
+			{
+				me->DisableCloak();
+			}
+
+			if (pPlayerToMirror->IsDucking())
+			{
+				me->PressCrouchButton(0.5f);
+			}
+			else
+			{
+				me->ReleaseCrouchButton();
+			}
+		}
+	}
+
+	// Calibrate dynamic follow distance
+	float follow_stop_distance_sq = sv_neo_bot_cmdr_stop_distance_sq.GetFloat();
+	if (pCommander->m_flBotDynamicFollowDistanceSq > follow_stop_distance_sq)
+	{
+		follow_stop_distance_sq = pCommander->m_flBotDynamicFollowDistanceSq;
+	}
+
+	if (pCommander->IsAlive())
+	{
```

Comment:
Nit: this condition goes 4 levels deep (if -> else if -> if -> if ...) which can make it a bit confusing to keep track of the scope. I'd personally prefer if we reversed this base case condition to an early return:
```cpp
if (!pCommander->IsAlive())
{
	// Commander died
	return false; // This triggers OnEnd which clears vars
}
```
so that we can collapse the lowest indent away entirely.